var/spack/repos/builtin/packages/dihydrogen/package.py

```diff
@@ -77,10 +77,16 @@ class Dihydrogen(CMakePackage, CudaPackage):
 
     # Add Aluminum variants
     depends_on('aluminum +cuda +nccl +ht +cuda_rma', when='+al +cuda')
+    depends_on('aluminum +rocm +rccl +ht +cuda_rma', when='+al +rocm')
 
     for arch in CudaPackage.cuda_arch_values:
         depends_on('aluminum cuda_arch=%s' % arch, when='+al +cuda cuda_arch=%s' % arch)
 
```

Comment:
should this have a `conflicts('+cuda', when='+rocm', msg='CUDA and ROCm support are mutually exclusive')` too?

Reply:
I don't know how to set a conflicts between the `cuda_arch` and `amdgpu_target` variants since they are not just boolean flags.  I would assume that I want something like `conflicts('cuda_arch~=none', when='amdgpu_target~=none')` but I am not sure that will work.

Reply:
I think there's already conflicts in CudaPackage and ROCmPackage that handle this; you can't set a cuda target when ~cuda it set etc.